node/src/actors/sessions_manager/mod.rs

```diff
@@ -136,10 +136,26 @@ impl SessionsManager {
                 command: SendGetPeers {},
                 safu: false,
             });
+
             act.discovery_peers(ctx, discovery_peers_period);
         });
     }
 
+    /// Method to periodically melt peers
+    fn melt_peers(&self, ctx: &mut Context<Self>, melt_peers_period: Duration) {
+        let peers_manager_addr = PeersManager::from_registry();
```

Comment:
Do we know if this call has any overhead? This function is somehow recursive, so we could pass `peers_manager_addr` as an `Option` argument and only do `from_registry` if it is `None`, so it's only called once.

Here of course I'm assuming that actors don't change addresses for the lifetime of the Actix `System`.

If on the contrary actors can change addresses, then there's a little problem with the current implementation because the address is obtained before `run_later`, but then used in the callback. During that time, `PeersManager` may have changed address, and therefore the message will likely go to "dead letter" and one iteration of `MeltPeers` will be missed. Not a big deal, I guess.

Speaking of what, I wonder if this is real recursion or `ctx.run_later` does actually use something like the actor mailbox at all. Why is this important? Taking into account that Rust doesn't have tail call optimization, if this is causing real recursion, we may easily exhaust the stack after a certain number of re-entries.

I imagine (but haven't confirmed at all) that clippy probably has some guards against dangerous recursion, but being Actix in the middle, I don't know if there could be something that it misses.